chemclipse/plugins/org.eclipse.chemclipse.msd.model/src/org/eclipse/chemclipse/msd/model/matrix/ExtractedMatrix.java

```diff
@@ -116,72 +115,49 @@ private int getMaxMz() {
 				.max(Comparator.comparing(IIon::getIon)) //
 				.get() //
 				.getIon();
-		int maxMz = (int)Math.round(max);
+		int maxMz = AbstractIon.getIon(max);
 		return (maxMz);
 	}
 
-	@Override
-	public float[] getMatrix() {
-		if(highRes == true) {
-			return null;			
-		}
+	public float[][] getMatrix() {
 		return this.signal;
 	}
 
-	@Override
 	public int getNumberOfScans() {
-		if(highRes == true) {
-			return 0;			
-		}
 		return this.numberOfScans;
 	}
 
-	@Override
 	public int getNumberOfIons() {
-		if(highRes == true) {
-			return 0;			
-		}
 		return this.numberOfIons;
 	}
 
-	@Override
-	public void updateSignal(float[] signal, int numberOfScans, int nubmerOfIons ) {
+	public void updateSignal(float[][] signal, int numberOfScans, int nubmerOfIons ) {
 		
 		IScanMSD currentScan;
 		IIon currentIon;
 		
-		if(highRes != true) {
-			try {
-				for(int i = 1; i <= numberOfScans; i++) {
-					currentScan = (IScanMSD) selection.getChromatogram().getScan(i);
-					currentScan.removeAllIons();
-					for(int j = startIon; j < stopIon; j++) {
-						if(signal[(j - startIon) * numberOfScans + (i-1)] != 0.0) {
-							currentIon = new Ion(j, signal[(j - startIon) * numberOfScans + (i - 1)] ); 
-							currentScan.addIon(currentIon);
-						}		
-					}
+		try {
+			for(int i = 1; i <= numberOfScans; i++) {
+				currentScan = (IScanMSD) selection.getChromatogram().getScan(i);
+				currentScan.removeAllIons();
+				for(int j = startIon; j < stopIon; j++) {
+					if(signal[i-1][j-startIon] != 0.0) {
+						currentIon = new Ion(j, signal[i - 1][j - startIon] ); 
+						currentScan.addIon(currentIon);
+					}		
 				}
-			} catch (Exception e) {
-				e.printStackTrace();
-			}			
-		}
+			}
+		} catch (Exception e) {
+			throw new RuntimeException(e);
```

Comment:
its a good idea to add a description here, at least something like "writing data back failed" as the exception message. In general its good not to simply catch "Exception" but narrow down the exception (types) as much as possible so you can give good hints to the user (e.g. "can't add ion")